python/cudf/cudf/core/column/categorical.py

```diff
@@ -1472,6 +1463,53 @@ def view(self, dtype: Dtype) -> ColumnBase:
             "Categorical column views are not currently supported"
         )
 
+    @staticmethod
+    def _concat(objs: MutableSequence[CategoricalColumn]) -> CategoricalColumn:
+        # TODO: This function currently assumes it is being called from
+        # column._concat_columns, at least to the extent that all the
+        # preprocessing in that function has already been done. That should be
+        # improved as the concatenation API is solidified.
+
+        # Find the first non-null column:
+        head = next((obj for obj in objs if obj.valid_count), objs[0])
+
+        # Combine and de-dupe the categories
+        cats = (
+            cudf.concat([o.cat().categories for o in objs])
+            .to_series()
+            .drop_duplicates(ignore_index=True)
+            ._column
+        )
```

Comment:
Should this be moved to after `newsize` is computed and checked? Seems redundant if `newsize == 0`.

Reply:
The output of this is used to modify `objs`, which are then eventually used in the computation of `newsize`. I don't know enough yet about how categories work to know whether these changes (mainly the calls to `_set_categories`) might affect the size, would our filter be too big or too small if we filtered before the conversion?

Reply:
You are right, it is used below.